core/trino-main/src/main/java/io/trino/execution/SqlTaskExecution.java

```diff
@@ -108,6 +110,8 @@
 
     // number of created PrioritizedSplitRunners that haven't yet finished
     private final AtomicLong remainingSplitRunners = new AtomicLong();
+    // number of created Drivers that haven't yet been destroyed
+    private final AtomicLong liveCreatedDrivers = new AtomicLong();
```

Comment:
nit: I would recommend encapsulating this counter in `DriverAndTaskTerminationTracker` and change the methods in `DriverAndTaskTerminationTracker`. For example:

- `DriverAndTaskTerminationTracker#driverCreated`
- `DriverAndTaskTerminationTracker#driverDestroyed`
- `DriverAndTaskTerminationTracker#getLiveDrivers` (for stats)
- `DriverAndTaskTerminationTracker#attemptTaskTermination` (or something along the lines)

Reply:
Good idea, will refactor that into the tracker instead.